src/transformers/trainer.py

```diff
@@ -3938,7 +3949,14 @@ def _save(self, output_dir: Optional[str] = None, state_dict=None):
             self.processing_class.save_pretrained(output_dir)
 
         # Good practice: save your training arguments together with the trained model
-        torch.save(self.args, os.path.join(output_dir, TRAINING_ARGS_NAME))
+        if not binary_serializiation:
+            self.args.to_json_file(os.path.join(output_dir, TRAINING_ARGS_NAME))
+            logger.info(
+                "safe serialization has been enabled by default and the training args will be stored in a {TRAINING_ARGS_NAME} \n"
+                "to fall back to the {DEPRECATED_ARGS_NAME} you can set os.environ['binary_serializiation']= '1' "
+            )
```

Comment:
Let's be a bit more explicit. Do a `warnings.warn()` & mention deprecation in 5.0.0 here please! :)